app/Http/Controllers/ResultController.php

```diff
@@ -116,40 +116,20 @@ public function index(Election $election)
         $tieBreakerBallot = $nBallots[$tieBreakerBallotIndex];
 
         $calculator = new RankedPairsCalculator($tieBreakerBallot);
-        $winnerOrder = $calculator->calculate($numWinners, ...$nBallots);
-        var_dump($winnerOrder);
-        return;
-
-        $pivotCandidates = array();
-        $tidemanCandidates = array();
-        // $election->
-        foreach ($election->candidates as $c) {
-            $id = $c["id"];
-            $pivotCandidates[$id] = $c;
-            $tidemanCandidates[$id] = new Candidate($id, $name = $c["name"]);
-        }
-
-        // construct an arbitrary tie-breaking ballot (TODO: do this in a non-arbitrary way)
-        $candidateLists = array_map(
-            function($c) {return new CandidateList($c);},
-            array_values($tidemanCandidates)
-        );
-        $tieBreaker = new Ballot(...$candidateLists);
-
-        // TODO: populate ballots with actual user rankings
-        $ballot = new NBallot(1, ...$candidateLists);
-        $ballots = [$ballot];
-
-        // compute election results
-        $instance = new RankedPairsCalculator($tieBreaker);
-        $winnerOrder = $instance->calculate(sizeof($tidemanCandidates), ...$ballots);
+        $winnerOrder = $calculator->calculate($numWinners, ...$nBallots)->toArray();
 
+        $pivotCandidates = $election->candidates()->get()->keyBy('id');
+        
         // format for return (i.e., convert tideman candidates back to pivot candidates)
-        $rv = array("order" => array());
-        foreach ($winnerOrder as $w) {
-            $candidate = $pivotCandidates[$w->getId()];
-            array_push($rv["order"], $candidate);
+        $orderedPivotCandidates = [];
+        //use explicit numerical indexing because order is very important
+        for($i = 0; $i < sizeof($winnerOrder); $i++) {
+            $tidemanCandidate = $winnerOrder[$i];
+            $candidateId = (int)$tidemanCandidate->getId();
+            $pivotCandidate = $pivotCandidates->get($candidateId);
+            $orderedPivotCandidates[] = $pivotCandidate;
```

Comment:
What is this syntax doing?  (for my information)

Reply:
Array append